### PR TITLE
Provide Array.asList instead of asIterable

### PR DESCRIPTION
**Pull request by piraces**
Resolves https://github.com/robstoll/atrium/issues/183

----
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

**Comment by piraces** [comment]
Hmm not sure why "codecov/patch" check is failing... Could you help me @robstoll ?

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/ArrayAsListAssertionsSpec.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package ch.tutteli.atrium.api.fluent.en_GB
+
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.specs.notImplemented
+
+object ArrayAsListAssertionsSpec : ch.tutteli.atrium.specs.integration.ArrayAsListAssertionsSpec (
+    "asIterable",
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/ArrayAsListAssertionsSpec.kt]:
```suggestion
    "asList",
```

File: domain/builders/atrium-domain-builders-common/src/main/kotlin/ch/tutteli/atrium/domain/builders/utils/nullable.kt [hunk 2]
```diff
@@ -28,16 +28,16 @@ inline fun <T> nullable(t: T): T? = t
 inline fun <T> nullable(t: KFunction0<T>): KFunction0<T?> = t
 
 /**
- * Turns an [Iterable] into an iterable with a nullable element type.
+ * Turns an [List] into an list with a nullable element type.
```
Comment (robstoll) [inline domain/builders/atrium-domain-builders-common/src/main/kotlin/ch/tutteli/atrium/domain/builders/utils/nullable.kt]: that's too much, also the other changes in this file

File: domain/builders/atrium-domain-builders-jvm/src/test/kotlin/ch/tutteli/atrium/domain/builders/utils/NullableSpec.kt [hunk 3]
```diff
@@ -13,9 +13,9 @@ object NullableSpec : Spek({
     //TODO use function reference directly once https://youtrack.jetbrains.com/issue/KT-24606 is fixed
     val t: (Int) -> Int? = ::nullable
     val nullableFun: KFunction1<Int, Int?> = t as KFunction1<Int, Int?>
-    val t2: (Iterable<Int>) -> Iterable<Int?> = ::nullableContainer
-    val nullableContainerFun: KFunction1<Iterable<Int>, Iterable<Int?>> =
-        t2 as KFunction1<Iterable<Int>, Iterable<Int?>>
+    val t2: (List<Int>) -> List<Int?> = ::nullableContainer
```
Comment (robstoll) [inline domain/builders/atrium-domain-builders-jvm/src/test/kotlin/ch/tutteli/atrium/domain/builders/utils/NullableSpec.kt]: that's too much, also the other changes until line 18

File: misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/ArrayAsListAssertionsSpec.kt [hunk 4]
```diff
@@ -5,69 +5,69 @@ import ch.tutteli.atrium.api.fluent.en_GB.containsExactly
 import ch.tutteli.atrium.api.verbs.internal.expect
 import ch.tutteli.atrium.creating.Expect
 import ch.tutteli.atrium.specs.*
-import ch.tutteli.atrium.translations.DescriptionIterableAssertion
+import ch.tutteli.atrium.translations.DescriptionListAssertion
```
Comment (robstoll) [inline misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/ArrayAsListAssertionsSpec.kt]: revert
```suggestion
import ch.tutteli.atrium.translations.DescriptionIterableAssertion
```

File: translations/en_GB/atrium-translations-en_GB-common/src/main/kotlin/ch/tutteli/atrium/translations/DescriptionListAssertion.kt [hunk 5]
```diff
@@ -7,6 +7,7 @@ import ch.tutteli.atrium.reporting.translating.StringBasedTranslatable
  * Contains the [DescriptiveAssertion.description]s of the assertion functions which are applicable to [Map].
  */
 enum class DescriptionListAssertion(override val value: String) : StringBasedTranslatable {
+    AN_ENTRY_WHICH_IS("an entry which is"),
```
Comment (robstoll) [inline translations/en_GB/atrium-translations-en_GB-common/src/main/kotlin/ch/tutteli/atrium/translations/DescriptionListAssertion.kt]: revert
```suggestion
```

File: misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/ArrayAsListAssertionsSpec.kt [hunk 6]
```diff
@@ -79,81 +79,81 @@ abstract class ArrayAsIterableAssertionsSpec(
     fun doubles(vararg doubles: Double) = doubles
     fun booleans(vararg booleans: Boolean) = booleans
 
-    val anEntryWhich = DescriptionIterableAssertion.AN_ENTRY_WHICH_IS.getDefault()
+    val anEntryWhich = DescriptionListAssertion.AN_ENTRY_WHICH_IS.getDefault()
```
Comment (robstoll) [inline misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/ArrayAsListAssertionsSpec.kt]: revert
```suggestion
    val anEntryWhich = DescriptionIterableAssertion.AN_ENTRY_WHICH_IS.getDefault()
```

File: domain/builders/atrium-domain-builders-common/src/main/kotlin/ch/tutteli/atrium/domain/builders/utils/nullable.kt [hunk 7]
```diff
@@ -35,7 +35,7 @@ inline fun <T> nullable(t: KFunction0<T>): KFunction0<T?> = t
  *
  * Basically it is a replacement for cast but without introducing one explicitly (we only give an additional hint to the
  * compiler that we expect a nullable type and not a non-nullable one). For instance, instead of writing
- * `getPersons() as List<String?>` you can write `nullableContainer(getPersons())`
+ * `getPersons() as Iterable<String?>` you can write `nullableContainer(getPersons())`
```
Comment (robstoll) [inline domain/builders/atrium-domain-builders-common/src/main/kotlin/ch/tutteli/atrium/domain/builders/utils/nullable.kt]: wouldn't have been necessary but is fine as well

Comment (piraces) [reply domain/builders/atrium-domain-builders-common/src/main/kotlin/ch/tutteli/atrium/domain/builders/utils/nullable.kt]: Thanks I'll keep it

**Comment by piraces** [comment]
Thank you for your feedback! I have applied all requested changes 😄

**Comment by robstoll** [comment]
@piraces thanks for the contribution :+1: 
Fancy another one?

**Comment by piraces** [comment]
Yeah! Why not? 😄 
I'll take a look at open issues tomorrow 👍

**Pull request merged**